assets/js/modules/analytics/common/ProfileNameTextField.js

```diff
@@ -54,21 +54,23 @@ export default function ProfileNameTextField() {
 	let helperText;
 	let trailingIcon;
 
-	const existingProfile = profiles.some( ( { name } ) => name === profileName );
```

Comment:
Nit-picking but we could probably simplify this PR to using an inline `Array.isArray( profiles )` here.

I think even better though it would be to return include `! Array.isArray( profiles )` (or `profiles === undefined` like we do elsewhere) in the clause above that returns `null`. The only reason that `profiles` would not be an array is that the data is not loaded yet, and for this kind of check we always have been using early returns, I think that clarifies it better.

Reply:
@felixarntz, we can't return if `profiles` is not an array because we need profiles just to determine whether or not we need to display a warning message when a user enters a duplicate name. I have updated my PR to have more elegant approach to determine if it is an existing profile or not.

Reply:
Oh you're right, I missed that. Looks good now!